### DISCUSSION TITLE
Empty `CollectionName` for record in hook

### DISCUSSION THREAD

**Discussion by jnsprnw** [pocketbase/pocketbase#6142] [Q&A] [answered]
Hey everyone,

Thank you so much for providing this library! It has been incredibly helpful in many of my projects.

I am trying to send a POST request every time a record is updated, and I'm using `onRecordAfterUpdateSuccess` for this purpose. The HTTP request itself is working fine, but I would like to include a descriptive title indicating what triggered the request. Specifically, I'd like to include the `collectionName`.

While I can access regular fields using `e.record.get('regularField')`, attempting to get the `collectionName` with `e.record.get('collectionName')` returns `null`. Interestingly, when I use `JSON.stringify(e.record)`, the `collectionName` appears in the output.

Is there a way to retrieve the `collectionName` directly?

Here's a snippet of my current code:

```js
onRecordAfterUpdateSuccess((e) => {
    console.log(JSON.stringify(e.record));
    console.log('regularField', e.record.get('regularField')); // Works
    console.log('collectionName', e.record.get('collectionName')); // null
    console.log('collectionId', e.record.get('collectionId')); // null
    console.log('collectionName', e.record.collectionName); // undefined

    const collectionName = e.record.get('collectionName');
    const triggerTitle = `Pocketbase+trigger+from+${collectionName}`; // Pocketbase+trigger+from+null

    console.log(triggerTitle);

    e.next();
});
```

**Comment by ganigeorgiev** [comment, marked as answer]
`collectionName` and `collectionId` are JSON response only fields and they are not part of the collection fields list.

If you want access to the record collection use `e.record.collection().name`.

_Side-note: you can add `/// <reference path="../pb_data/types.d.ts" />` at the top of your hooks file so that your editor can assist you with autocompletion and documentation on hover. For all available methods and fields, you can also check https://pocketbase.io/jsvm/interfaces/core.Record.html._

**Reply by jnsprnw** [reply]
Wow, that was incredibly fast! Thank you so much!

For anyone interested in doing something similar, here is my complete code for triggering a Netlify hook:

```js
onRecordAfterUpdateSuccess((e) => {
	// Attempting to retrieve the collection name
	const collectionName = e.record.collection().name;
	const triggerTitle = `Pocketbase+trigger+from+${collectionName}`;

	try {
		// Sending a POST request to the Netlify webhook URL
		const res = $http.send({
			url: `https://api.netlify.com/build_hooks/xxx?trigger_title=${triggerTitle}`, // Replace xxx with your hook id
			method: 'POST',
			headers: { 'Content-Type': 'application/json' }
		});

		// Check if the response status is 200 (successful)
		if (res.statusCode === 200) {
			console.log('Successfully triggered Netlify');
		}
	} catch (error) {
		// Log any errors during the request
		console.error('Error executing Netlify hook:', error);
	}

	// Continue to the next middleware or function
	e.next();
});
```
